File: JCRazor/wwwroot/CssStyleOperator.js
export class CssStyleOperator
{
	/**
	 * 
	 * @param {HTMLElement} element
	 * @param {string} style_name
	 * @returns
	 */
	static get_computed_style(element, style_name)
	{
		console.log(window.getComputedStyle(element));
		return window.getComputedStyle(element)[style_name];
	}

	/**
	 * 
	 * @param {HTMLElement} element
	 * @param {string} width
	 */
	static set_width(element, width)
	{
		element.style.width = width;
	}

	/**
	 * 
	 * @param {HTMLElement} element
	 * @param {string} height
	 */
	static set_height(element, height)
	{
		element.style.height = height;
	}

	/**
	 * 
	 * @param {HTMLElement} element
	 * @param {string} padding
	 */
	static set_padding(element, padding)
	{
		element.style.padding = padding;
	}

	/**
	 * 
	 * @param {HTMLElement} element
	 * @param {string} margin
	 */
	static set_margin(element, margin)
	{
		element.style.margin = margin;
	}
}

File: JCRazor/js互操作/CssStyleOperator.cs
using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

namespace JCRazor.js互操作;

public class CssStyleOperator
{
	public CssStyleOperator(IJSRuntime jsrt)
	{
		_jsm = new JSModule(jsrt, "./_content/JCRazor/CssStyleOperator.js");
	}

	private JSModule _jsm;

	public async Task<string> GetComputedStyleAsync(ElementReference element, string style_name)
	{
		return await _jsm.InvokeAsync<string>("CssStyleOperator.get_computed_style", element, style_name);
	}

	public async Task SetWidthAsync(ElementReference element, string width)
	{
		await _jsm.InvokeVoidAsync("CssStyleOperator.set_width", element, width);
	}

	public async Task SetHeightAsync(ElementReference element, string height)
	{
		await _jsm.InvokeVoidAsync("CssStyleOperator.set_height", element, height);
	}

	public async Task SetPaddingAsync(ElementReference element, string padding)
	{
		await _jsm.InvokeVoidAsync("CssStyleOperator.set_padding", element, padding);
	}

	public async Task SetMarginAsync(ElementReference element, string margin)
	{
		await _jsm.InvokeVoidAsync("CssStyleOperator.set_margin", element, margin);
	}
}
